### DISCUSSION TITLE
Change BOOTSEL baud rate to something other than 1200

### DISCUSSION THREAD

**Discussion by RhinoLance** [earlephilhower/arduino-pico#2272] [Q&A] [answered]
I'm working on a project which needs to support a protocol requiring connecting at 1200 baud.  This of course clashes with arduino-pico's boot-select protocol which also needs 1200 baud.

I have manually changed the required line in SerialUSB.cpp from
if (!_rebooting && (_bps == **1200**) && (!_dtr)) {
to
if (!_rebooting && (_bps == **300**) && (!_dtr)) {

While this works, programming the board is a pain as I now need to unplug, press BOOTSEL, plug back in .... . 

Is there a simpler way to change the magic number of 1200 baud, and still have the build toolchain function?

Ta,
  Lance

**Comment by earlephilhower** [comment, marked as answer]
Sorry, no.  That's an Arduino "standard" and implemented in many cores.

Since this only affects USB, and the "baud" rate is just a number and not actually used anywhere, though, why is this an issue?  A USB CDC port set to "115200bps" can transfer the same amount of data as one set to "110bps".  Only on the actual UARTS (Serial!/2) will the baud rate set actually make a difference, and those are *not* checked for Arduino reset protocol.

**Comment by RhinoLance** [comment]
Thanks for your response.

Ironically, the protocol I'm implementing is for hardware which supports two modes of operation, with the mode dictated by the connection rate.   In effect doing the same thing as what the boot loader select is doing.  It's old hardware from decades before arduino was a thing.  Just an unfortunate coincidence.

**Reply by JAndrassy** [reply]
and it has USB?

**Reply by RhinoLance** [reply]
I'm currently using the USB port on the Pico for programming and running.  I could use one of the UARTs, but that introduces other issues in the environment.  

If there was a simple workaround it would have been nice to know, but as there isn't, it's really not that big a deal.  I've started keeping a terminal window open, so that i can fire off a simple telnet connection to trigger the BOOTSEL.

**Reply by JAndrassy** [reply]
how does the USB port of the Pico related to the 1200 baud of the device?

**Reply by RhinoLance** [reply]
The Pico has replaced the computer interface part of the device.  The control software running on a computer sets the device's mode of operation by the baud rate it connects at.  The control software is code over which I have no control, and the odds of it ever changing are slim to none.

Did that answer your question, or an I misunderstanding what you're asking?

**Reply by JAndrassy** [reply]
ok sorry I thought the Pico communicates with the device
